From 9b5525a7adec091df713bc1fd71eae678b0d20e8 Mon Sep 17 00:00:00 2001
From: Damien Riegel <damien.riegel@savoirfairelinux.com>
Date: Mon, 2 Nov 2015 09:43:21 -0500
Subject: [PATCH] ts4800: patch env settings to boot dts

Signed-off-by: Damien Riegel <damien.riegel@savoirfairelinux.com>
---
 include/configs/ts4800.h | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/include/configs/ts4800.h b/include/configs/ts4800.h
index 1624a1b..f096789 100644
--- a/include/configs/ts4800.h
+++ b/include/configs/ts4800.h
@@ -111,19 +111,23 @@
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"script=boot.scr\0" \
-	"image=uImage\0" \
+	"image=zImage\0" \
+	"fdt=imx51-ts4800.dtb\0" \
+	"kernel_addr=0x90008000\0" \
+	"fdt_addr=0x91000000\0" \
 	"mmcdev=0\0" \
-	"mmcpart=1\0" \
-	"mmcargs=setenv bootargs root=/dev/mmcblk0p2 rootwait rw\0" \
+	"mmcpart=2\0" \
+	"mmcargs=setenv bootargs root=/dev/mmcblk0p3 rootwait rw\0" \
 	"addtty=setenv bootargs ${bootargs} console=ttymxc0,${baudrate}\0" \
 	"loadbootscript=" \
 		"fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
 		"source\0" \
-	"loadimage=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${image};\0" \
+	"loadimage=fatload mmc ${mmcdev}:${mmcpart} ${kernel_addr} ${image}; " \
+		  "fatload mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${fdt}\0" \
 	"mmcboot=echo Booting from mmc ...; " \
 		"run mmcargs addtty; " \
-                "bootm; "
+                "bootz ${kernel_addr} - ${fdt_addr}; "
 
 #define CONFIG_BOOTCOMMAND \
 	"mmc dev ${mmcdev}; if mmc rescan; then " \
-- 
2.5.0

